vms/platformvm/utxo/camino_locked.go

```diff
@@ -1098,53 +1089,43 @@ func (h *handler) VerifyUnlockDepositedUTXOs(
 
 		out := utxo.Out
 		lockIDs := &locked.IDsEmpty
-		isDeposited := false
 		if lockedOut, ok := out.(*locked.Out); ok {
 			// utxo isn't deposited, so it can't be unlocked
 			// bonded-not-deposited utxos are not allowed
-			if isDeposited = lockedOut.DepositTxID != ids.Empty; !isDeposited {
-				return nil, errUnlockingUnlockedUTXO
+			if lockedOut.DepositTxID == ids.Empty {
+				return errUnlockingUnlockedUTXO
 			}
 			out = lockedOut.TransferableOut
 			lockIDs = &lockedOut.IDs
 		}
 
 		in := input.In
 		if lockedIn, ok := in.(*locked.In); ok {
-			// This input is locked, but its LockIDs is wrong
+			// Input is locked, but LockIDs isn't matching utxo's
 			if *lockIDs != lockedIn.IDs {
-				return nil, errLockIDsMismatch
+				return errLockIDsMismatch
 			}
 			in = lockedIn.TransferableIn
 		} else if lockIDs.IsLocked() {
-			// The UTXO says it's locked, but this input, which consumes it,
-			// is not locked - this is invalid.
-			return nil, errLockedFundsNotMarkedAsLocked
+			// The UTXO is locked, but consuming input isn't locked
+			return errLockedFundsNotMarkedAsLocked
 		}
 
 		consumedAmount := in.Amount()
 
-		if isDeposited {
-			// verifying that input amount equal to utxo amount
-			if innerOut, ok := out.(*secp256k1fx.TransferOutput); !ok || innerOut.Amt != consumedAmount {
-				return nil, fmt.Errorf("failed to verify transfer: %w", errUTXOOutTypeOrAmtMissmatch)
-			}
-
-			deposit, err := chainState.GetDeposit(lockIDs.DepositTxID)
-			if err != nil {
-				return nil, err
-			}
-
-			if !deposit.IsExpired(currentTimestamp) {
-				if err := h.fx.VerifyMultisigTransfer(tx, in, creds[index], out, h.utxosReader); err != nil {
-					return nil, fmt.Errorf("failed to verify transfer: %w", err)
-				}
+		if verifyCreds {
+			if err := h.fx.VerifyMultisigTransfer(tx, in, creds[index], out, h.utxosReader); err != nil {
+				return fmt.Errorf("failed to verify transfer: %w: %s", errCantSpend, err)
 			}
+		} else if innerOut, ok := out.(*secp256k1fx.TransferOutput); !ok || innerOut.Amt != consumedAmount {
+			return fmt.Errorf("failed to verify transfer: %w", errUTXOOutTypeOrAmtMissmatch)
```

Comment:
Typo: errUTXOOutTypeOrAmtMissmatch -> errUTXOOutTypeOrAmtMismatch (Its also used in other places)